### PR TITLE
Add `module_path` to `ContractArtifacts`

### LINKED ISSUE #1048: Add contract module path information to Starknet artefacts file
### Problem

Foundry needs to know the Cairo source module path of compiled contracts it is operating on. 

### Proposed Solution

Let's provide them this information, by writing this path to the Starknet artefacts file.

Example:

```json
{
  "version": 1,
  "contracts": [
    {
      "id": "<opaque>",
      "package_name": "mypackage",
      "contract_name": "Contract2",
      "module_path": "mypackage::path::to::module::Contract2",
      "artifacts": {
        "sierra": "mypackage_Contract2.contract_class.json",
        "casm": null
      }
    },
    {
      "id": "<opaque>",
      "package_name": "mypackage",
      "contract_name": "Contract1",
      "module_path": "mypackage::Contract1",
      "artifacts": {
        "sierra": "mypackage_Contract1.contract_class.json",
        "casm": null
      }
    }
  ]
}
```

### Notes

Implementation starting points:
https://github.com/software-mansion/scarb/blob/b3b65fc08e2a55353751370882ef26ba8c112d20/scarb/src/compiler/compilers/starknet_contract.rs#L162-L168
https://github.com/software-mansion/scarb/blob/b3b65fc08e2a55353751370882ef26ba8c112d20/scarb/src/compiler/compilers/starknet_contract.rs#L260-L304

### PR DESCRIPTION
**Pull request by tomek0123456789**
Closes #1048

### CODE REVIEW

File: scarb/src/compiler/compilers/starknet_contract.rs [hunk 1]
```diff
@@ -216,6 +223,14 @@ impl Compiler for StarknetContractCompiler {
...
 
         let main_crate_ids = collect_main_crate_ids(&unit, db);
+        // main crate ids to definicja ze scarb.tomla glownego?
+        eprintln!("unit = {:#?}", unit);
+        eprintln!("unit.main_component() = {:#?}", unit.main_component());
+        eprintln!(
+            "unit.main_component().cairo_package_name() = {:#?}",
+            unit.main_component().cairo_package_name()
+        );
+        eprintln!("main_crate_ids = {:#?}", main_crate_ids);
```
Comment (mkaput) [inline scarb/src/compiler/compilers/starknet_contract.rs:233]:
```suggestion
```

btw https://doc.rust-lang.org/stable/std/macro.dbg.html

File: scarb/tests/build_starknet_contract.rs [hunk 2]
```diff
@@ -5,6 +5,7 @@ use cairo_lang_starknet::contract_class::ContractClass;
 use indoc::{formatdoc, indoc};
 use itertools::Itertools;
 use predicates::prelude::*;
+use std::ops::Deref;
```
Comment (mkaput) [inline scarb/tests/build_starknet_contract.rs:8]: 🤔 what is this for?

**Review by mkaput — CHANGES_REQUESTED** [review]

File: scarb/src/compiler/compilers/starknet_contract.rs (hunk 1 above)
Comment (tomek0123456789) [reply scarb/src/compiler/compilers/starknet_contract.rs:233]: woops, was almost sure that i didn't push that, my bad 😅

File: scarb/tests/build_starknet_contract.rs (hunk 2 above)
Comment (tomek0123456789) [reply scarb/tests/build_starknet_contract.rs]: 🤫

File: scarb/src/compiler/compilers/starknet_contract.rs [hunk 3]
```diff
@@ -164,15 +164,22 @@ struct ContractArtifacts {
...
+        package_name: &PackageName,
+        contract_name: &str,
+        contract_path: &str,
+        module_path: String,
+    ) -> Self {
         Self {
             id: short_hash((&package_name, &contract_path)),
             package_name: package_name.clone(),
             contract_name: contract_name.to_owned(),
+            module_path,
```
Comment (mkaput) [inline scarb/src/compiler/compilers/starknet_contract.rs:182]:
```suggestion
        module_path: &str,
    ) -> Self {
        Self {
            id: short_hash((&package_name, &contract_path)),
            package_name: package_name.clone(),
            contract_name: contract_name.to_owned(),
            module_path: module_path.to_owned(),
```

File: scarb/src/compiler/compilers/starknet_contract.rs [hunk 4]
```diff
@@ -275,6 +282,7 @@ impl Compiler for StarknetContractCompiler {
                 &package_name,
                 &contract_name,
                 contract_selector.full_path().as_str(),
+                decl.module_id().full_path(db.upcast_mut()).clone(),
```
Comment (mkaput) [inline scarb/src/compiler/compilers/starknet_contract.rs:285]:
```suggestion
                &decl.module_id().full_path(db.upcast_mut()),
```

**Review by mkaput — COMMENTED** [review]
Shifting clones further would be more idiomatic Rust.

**Review by mkaput — APPROVED** [review]

**Pull request merged**